dev/com.ibm.ws.concurrent.mp_fat/test-applications/MPConcurrentCDIApp/src/concurrent/mp/fat/cdi/web/MPConcurrentCDITestServlet.java

```diff
@@ -0,0 +1,222 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package concurrent.mp.fat.cdi.web;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.inject.Inject;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.servlet.annotation.WebServlet;
+
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ManagedExecutorConfig;
+import org.eclipse.microprofile.concurrent.ThreadContext;
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+
+@SuppressWarnings("serial")
+@WebServlet(urlPatterns = "/MPConcurrentCDITestServlet")
+public class MPConcurrentCDITestServlet extends FATServlet {
+
+    static final int TIMEOUT_MIN = 2;
+
+    @Inject
+    ConcurrencyBean bean;
+
+    @Inject
+    ManagedExecutor noAnno;
+
+    @Inject
+    @ManagedExecutorConfig
+    ManagedExecutor defaultAnno;
+
+    @Inject
+    @ManagedExecutorConfig(maxAsync = -1, maxQueued = -1, propagated = ThreadContext.ALL_REMAINING, cleared = ThreadContext.TRANSACTION)
+    ManagedExecutor defaultAnnoVerbose;
+
+    @Inject
+    @ManagedExecutorConfig(maxAsync = 5)
+    ManagedExecutor maxAsync5;
+
+    @Inject
+    @ManagedExecutorConfig(maxAsync = 2, maxQueued = 2)
+    ManagedExecutor max2;
+
+    @Inject
+    @ManagedExecutorConfig(propagated = {}, cleared = { ThreadContext.APPLICATION })
+    ManagedExecutor noAppCtx;
+
+    @Inject
+    @ManagedExecutorConfig(propagated = { ThreadContext.APPLICATION, ThreadContext.TRANSACTION }, cleared = {})
```

Comment:
Liberty will not (at least initially) support propagating TRANSACTION context.  If this is permitted, we need to figure out where error checking is missing.  The same applies to the next injection point as well.
Also, correct spelling for field name is propagated.